### PR TITLE
Add Review API, rebuild serverCalls

### LINKED ISSUE #158: Refactor LeaveReview files

### PR DESCRIPTION
**Pull request by johndavis92790**

### CODE REVIEW

**Review by Kimberly-Rodriguez — DISMISSED** [review]

File: client/src/utils/helpers.tsx [hunk 1]
```diff
@@ -23,3 +23,13 @@ export function isEmail(str: string): boolean {
     /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
   return emailRegex.test(str);
 }
+
+export const STAR_RATING_NAMES = [
```
Comment (robyng) [inline client/src/utils/helpers.tsx:27]: Okay cool I see where you put the allergy option ratings now.

File: package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "lint": "tslint -p .",
     "gcp-build": "cd client && tsc -p .",
     "deploy": "gcloud app deploy",
-    "pretty": "prettier --write \"./**/*.{tsx,css,js,json}\""
+    "pretty": "prettier --write \"./**/*.{tsx,css,js,json}\"",
+    "serverOnly": "cd server && npm start"
```
Comment (robyng) [inline package.json:25]: Okay I will do npm i after this gets merged

**Review by robyng — CHANGES_REQUESTED** [review]
Hi it looks good except when I click create review it give a type error: 

TypeError: Cannot read properties of undefined (reading '0')
CreateReview
C:/Users/Robyn Graham/Desktop/dragon-fruit/Healthy-Eating-App/client/src/pages/Review/LeaveReview/CreateReview.tsx:145
  142 | return (
  143 |   <>
  144 |     <GooglePhoto
> 145 |       photo_reference={restaurantData.photos[0].photo_reference}
      | ^  146 |       max_height="200"
  147 |       max_width="200"

File: package.json (hunk 2 above)
Comment (johndavis92790) [reply package.json:25]: It runs `npm i` every time you run `npm run develop` automatically so don't worry about that.

File: client/src/utils/helpers.tsx (hunk 1 above)
Comment (johndavis92790) [reply client/src/utils/helpers.tsx:27]: Yep so all you have to do to change, add or remove these is change it here in this array, super easy!

**Comment by johndavis92790** [comment]
> Hi it looks good except when I click create review it give a type error:
> 
> TypeError: Cannot read properties of undefined (reading '0') CreateReview C:/Users/Robyn Graham/Desktop/dragon-fruit/Healthy-Eating-App/client/src/pages/Review/LeaveReview/CreateReview.tsx:145 142 | return ( 143 | <> 144 | <GooglePhoto
> 
> > 145 |       photo_reference={restaurantData.photos[0].photo_reference}
> > | ^  146 |       max_height="200"
> > 147 |       max_width="200"

Fixed this by adding a spinner and removing it when the restaurantData was loaded.

**Pull request merged**